datahub-web-react/src/app/entity/shared/containers/profile/header/EntityName.tsx

```diff
@@ -37,10 +39,32 @@ function EntityName(props: Props) {
 
     const [updateName] = useUpdateNameMutation();
 
+    const updateEntity = useEntityUpdate<GenericEntityUpdate>();
+
+    const updateEditableName = (entityNameUpdate, mutationUrn) => {
```

Comment:
Minor comment here: Name is a very important concept. It's important that we keep the APIs standard across _all_ entity types when manipulating this field. I would strongly request that we use the more structured "updateName" mutation to handle updates to _any_ entity's name. This is similar to what we do with the `updateDescription` mutation, which abstracts away description updates for each type of entity

What do you think?

Reply:
The reason this was built this way is to support being able to allow updating the editableDatasetProperties' Name field (and keeping the non-editable Name field intact). The current updateName mutation does not support Dataset entity name updates as I can understand ([/datahub/graphql/resolvers/mutate/UpdateNameResolver.java](https://github.com/datahub-project/datahub/blob/master/datahub-graphql-core/src/main/java/com/linkedin/datahub/graphql/resolvers/mutate/UpdateNameResolver.java#L48)).

If this is agreeable, I can extend the existing UpdateNameResolver to support Dataset entity's EditableDatasetProperties' Name field. Please advise and I can update it accordingly and abstract it from the UI perspective by just calling "updateName" mutation.

Reply:
I have made this change with the latest commit. Please look into.